### PR TITLE
feat(liquidity-mining): liquidity rolling logic with split pools

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

The previous LM roll, as documented in PR #1929, considered two distinct periods in the weekly LM rewards:

1) a period considering two pools during the roll &
2) a period considering the new pool after the roll.

This worked well as the rolling started at the beginning of the week, meaning there was only two distinct periods to consider.

For this previous LM rewards we calculated the payouts by a) generating the rolled payouts, b) generating the following week's payouts and c) joining the payouts. 

For this week's LM rewards things are a bit more complex. We now need to consider three periods:
1) a period of the "old" pool, running from 21 Sept 23 Sept,
2) a period considering both pools, during the roll, running from 23 Sept to 27 Sept &
3) a period considering the "new" pool running from 27 Sept to 28 Sept.

The new period this month is due to the timing of the roll not aligning with the start (or end) of given LM rewards.

For this week this looks like this:
Untitled-Page-2

**Summary**

**This PR adds a script to consider normal LM rewards (not rolled) that are split over two pools**. This script can then be used in conjunction with the previous roll and join scripts to generate this weeks payouts. 

**Details**

This new script takes in a bunch of parameters, as follows:
```
fromBlock1, 	// defines start of the first pool period
toBlock1, 		// defines end of the first pool period
fromBlock2,		// defines start of the second pool period
toBlock2,		// defines end of the second pool period
tokenName,
pool1Address,
pool2Address,
week,
umaPerPeriod1,	// number of rewards to send in the first period
umaPerPeriod2,	// number of rewards to send for the second period
blocksPerSnapshot
```

Subsequent PRs will contain the associated payouts.

### CODE REVIEW

File: packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
+require("dotenv").config();
+const fs = require("fs");
+const path = require("path");
+const Web3 = require("web3");
+const poolAbi = require("../../build/contracts/ERC20.json");
+const { _fetchBalancerPoolInfo, _calculatePayoutsBetweenBlocks } = require("./CalculateBalancerLPRewards"); // re-use balancer query function.
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js:6]: re-use basically all the same logic from `CalculateBalancerLPRewards`

File: packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    bPool2,
+    shareHolders2,
+    fromBlock2,
+    toBlock2,
+    blocksPerSnapshot,
+    umaPerSnapshot2,
+    snapshotsToTake2
+  );
+
+  const joinedShareHolderPayout = await _joinPayoutPeriods(shareHolderPayout1, shareHolderPayout2);
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js:119]: generate two sets of payouts for each pool then join them and save the output.

File: packages/core/scripts/liquidity-mining/JoinRolledPayouts.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ async function JoinRolledPayouts(week, rollNum, tokenName) {
     throw new Error("Missing or invalid parameter! Provide week, rollNum & tokenName");
```
Comment (chrismaree) [inline packages/core/scripts/liquidity-mining/JoinRolledPayouts.js:23]: small logic change in the joiner to consider the different ways that the script can be used

File: packages/core/scripts/liquidity-mining/JoinRolledPayouts.js [hunk 4]
```diff
@@ -39,9 +39,17 @@ async function JoinRolledPayouts(week, rollNum, tokenName) {
...
+  if (weeklyRewards.poolAddress && rollRewards.pool2Address != weeklyRewards.poolAddress)
     throw "The second rolled pool address must equal the incoming weekly rewards pool";
-
+  else if (
+    weeklyRewards.splitPeriodPool1Address &&
+    weeklyRewards.splitPeriodPool2Address &&
+    (rollRewards.pool1Address != weeklyRewards.splitPeriodPool1Address ||
+      rollRewards.pool2Address != weeklyRewards.splitPeriodPool2Address)
+  )
+    throw "The first pool address in the roll must equal the first pool in the split period and the second pool in the roll must equal the second split period.";
```
Comment (nicholaspai) [inline packages/core/scripts/liquidity-mining/JoinRolledPayouts.js:52]: Nice to have this test in the script

**Review by nicholaspai — APPROVED** [review]
Pretty complex logic, could you help point out where in particular the added complexity is? Overall LGTM I mostly looked for silly errors like replacing any of the _1 variables with a _2 variable

**Comment by daywiss** [comment]
looks good though i dont fully understand why a special script is needed vs summing the outputs of the scripts run independently for each time period/pool. Is this an optimization or do the number need to be treated differently?

**Review by daywiss — APPROVED** [review]

**Comment by chrismaree** [comment]
> Pretty complex logic, could you help point out where in particular the added complexity is? Overall LGTM I mostly looked for silly errors like replacing any of the _1 variables with a _2 variable

The nice thing about this script is that there is not really that much logic other than the point where the payouts are added together. I've pointed to that specific place in the code with a comment.

File: packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js (hunk 2 above)
Comment (chrismaree) [reply packages/core/scripts/liquidity-mining/CalculateSplitPoolBalancerLPRewards.js:119]: @nicholaspai this function has the main complexity.

**Comment by chrismaree** [comment]
> looks good though i dont fully understand why a special script is needed vs summing the outputs of the scripts run independently for each time period/pool. Is this an optimization or do the number need to be treated differently?

The reason I wrote a new script for this is that I wanted to keep the outputs in a consistent format with the past 8 weeks LM rewards. We could have easily run the normal script twice on each period and then add the outputs together. However, this does not conform to the standard we've followed to this point. For this reason, I created this new script to join them after creating the respective payouts for the 1st and 3rd payout periods.

**Pull request merged**